app/modules/projects/models.py

```diff
@@ -92,6 +97,36 @@ def add_encounter(self, encounter):
             db.session.add(enrollment)
             self.encounter_members.append(enrollment)
 
+    def add_user_in_context(self, user):
```

Comment:
These look good, let's call them from inside the `add_user` function as:
```
def add_user(self, user):
    with db.session.begin():
        self.add_user_in_context(user)
```
and so on for the others.